workflows/mag.nf

```diff
@@ -581,45 +603,70 @@ workflow MAG {
         // If any two of the binners are both skipped at once, do not run because DAS_Tool needs at least one
         if ( params.refine_bins_dastool ) {
 
-            BINNING_REFINEMENT ( BINNING_PREPARATION.out.grouped_mappings, BINNING.out.bins, BINNING.out.metabat2depths, ch_short_reads )
+            ch_refined_bins = Channel.empty()
+
+            if(params.bin_domain_classification) {
+                // bins which cannot be classified ('unknown') get fed into binning refinement
+                // but also could putatively be eukaryotic so keep separate as well
+                ch_prokarya_bins_dastool = ch_binning_results_bins
+                    .filter { meta, bins ->
+                        meta.domain in ["bacteria", "archaea", "prokarya", "organelle", "unknown"]
+                    }
+
+                ch_eukarya_bins_dastool = ch_binning_results_bins
+                    .filter { meta, bins ->
+                        meta.domain in ["eukarya", "unknown"]
```

Comment:
Wont there now be two copies of bins with `meta.domain == "unknown"` in `ch_refined_bins` because both, `ch_prokarya_bins_dastool` and `ch_eukarya_bins_dastool`, contain "unknown" and will be mixed in `ch_refined_bins = ch_eukarya_bins_dastool.mix(BINNING_REFINEMENT.out.refined_bins)`?

Reply:
I think that depends on the outcome of DASTool - here, we assume that an 'unknown' bin could be putatively eukaryotic or prokaryotic. On the assumption that it's eukaryotic, we keep it as is, and on the assumption it's prokaryotic, it gets fed into DASTool. Whether we get the same bin out at the end of the DASTool process depends on what's in the prokaryotic bins - I presume a bin that's 'unknown' is probably pretty poor quality anyway, so it might get chopped up and moved around?